astropy/coordinates/baseframe.py

```diff
@@ -617,6 +612,31 @@ def data(self):
                              'data'.format(repr(self)))
         return self._data
 
+    def update_representation_data(self, representation_data):
+        """
+        Update the data in this frame.
+
```

Comment:
Can you add something about restrictions?  It seems to me that the only "safe" way to use this is to have the new `representation_data` be the same class and shape as the existing frame data.  Well, it *might* be safe to do otherwise if the `_rep_cache` is cleared, but that would need to be proved with tests.